File: другое/тинькофф.аисд/2/E.ts
let n = 4;
let lst = [4, 1, 3, 2];

const stack: number[] = [];
let cur = 0;
const ans: [number, number][] = [];
let k = 0;
while (lst.length > 0) {
  const item = lst.shift()!;
  stack.push(item);
  k++;
  if (cur + 1 === stack[stack.length - 1]) {
    ans.push([1, k]);
    k = 0;
  }
  let flag = false;
  while (cur + 1 === stack[stack.length - 1]) {
    flag = true;
    k++;
    cur = stack.pop()!;
  }
  if (flag) {
    ans.push([2, k]);
    k = 0;
  }
}
console.log(ans);

if (stack.length === n) {
  console.log(ans.length);
  for (const item of ans) {
    console.log(item.join(" "));
  }
} else {
  console.log(0);
}

File: другое/тинькофф.аисд/2/E.py
n = int(input())
lst = list(map(int, input().split()))

stack = []
cur = 0
ans = []
k = 0

while len(lst) > 0:
    item = lst.pop(0)
    stack.append(item)
    k += 1
    if len(stack) > 0 and cur + 1 == stack[-1]:
        ans.append([1, k])
        k = 0
    flag = False
    while len(stack) > 0 and cur + 1 == stack[-1]:
        flag = True
        k += 1
        cur = stack.pop()
    if flag:
        ans.append([2, k])
        k = 0


if cur == n:
    print(len(ans))
    for item in ans:
        print(" ".join(map(str, item)))
else:
    print(0)
